### PR TITLE
Basicfeature

### PR DESCRIPTION
**Pull request by gautamvij**
updated directory structure and added search feature (Search feature will be updated later with other features in sync)

### CODE REVIEW

**Comment by pranjalk** [comment]
Add tech stack in readme
give commands directly for npm and node instead of giving links

File: .jshintrc [hunk 1]
```diff
@@ -0,0 +1,91 @@
+{
```
Comment (shreedharshetty) [inline .jshintrc]: why are you integrating jshintrc? no please
https://github.com/practo/javascript/blob/master/guides/coding-standards/tutorial.md

File: server/routes.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+              req.session.cookie.expires = false;
+            }
+        res.redirect('/');
+    });
+
+};
+
+function merge_options(obj1,obj2){
+	'use strict';
+    var obj3 = {};
```
Comment (shreedharshetty) [inline server/routes.js]: obj1, obj2 and obj3? where do you get such names from?

File: server/server.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+require('./routes')(app,passport);
+
+var models = require('./models/');
+
+models.sequelize.sync({
+	force: false,
+	logging: console.log
+}).then(function() {
+	'use strict';
```
Comment (shreedharshetty) [inline server/server.js]: why is `use strict` here? seriously?

File: server/server.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+	var server = app.listen(8080, function () {
+	    var host = server.address().address;
+	    var port = server.address().port;
+	    console.log('AnimeTracker is listening at http://%s:%s', host, port);
+	  });
+});
+// 	models.User.create({username: "pranjal", password : "hello"}).then(function(user1){
```
Comment (shreedharshetty) [inline server/server.js]: remove the deadcode

File: src/controllers/AnimeController.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+animetracker.controller('AnimeController', ['$scope', 'apiservice', '$routeParams', function ($scope, apiservice, $routeParams) {
+    'use strict';
+    var animeId = $routeParams.id;
+    
+    apiservice.getAnime(animeId).then(
+    function(payLoad){
```
Comment (shreedharshetty) [inline src/controllers/AnimeController.js]: this payload? shouldnt this be response? please name variable according to the context

File: src/css/components/_components.scss [hunk 6]
```diff
@@ -0,0 +1,23 @@
+
+.plus {
+   background: #F65A5B ;
```
Comment (shreedharshetty) [inline src/css/components/_components.scss]: where are the variables? is this how you write scss?

File: .jshintrc [hunk 7]
```diff
@@ -1,3 +1,91 @@
 {
-	"esversion": 6
+    // JSHint Default Configuration File (as on JSHint website)
```
Comment (pranjalk) [inline .jshintrc]: dont use jshint

File: package.json [hunk 8]
```diff
@@ -17,7 +17,8 @@
     "sequelize": "^3.24.1"
   },
   "devDependencies": {
-    "eslint": "^3.3.1"
+    "eslint": "^3.3.1",
+    "jshint": "^2.9.3"
```
Comment (pranjalk) [inline package.json]: remove all dependencies/dev-dependencies which are not being used

File: server/routes.js [hunk 9]
```diff
@@ -1,19 +1,63 @@
...
+			    	res.send(body);
+			      }
+			      else
+			      {
+			      	res.send(error);
+			      }
+			});
+	});
+
+	app.get('/addData', isLoggedIn, function(req,res){
```
Comment (pranjalk) [inline server/routes.js]: endpoint always all smallcase...

File: server/routes.js [hunk 10]
```diff
@@ -1,19 +1,63 @@
...
+		.then(function(store){
+			console.log(store);
+			res.send("1");
+		}).catch(function(error){
+			res.send(error);
+		});
+
+	});
+
+	app.get('/getData', isLoggedIn, function(req,res){
```
Comment (pranjalk) [inline server/routes.js]: again small case

File: server/routes.js [hunk 11]
```diff
@@ -89,18 +140,19 @@ module.exports = function (app,passport){
 };
 
 function merge_options(obj1,obj2){
```
Comment (pranjalk) [inline server/routes.js:159]: functions in camelCase

File: src/css/main.css [hunk 12]
```diff
@@ -1,16 +1,82 @@
-#header {
-  position: absolute;
-  top: 0;
-  left: 0;
+/**
```
Comment (pranjalk) [inline src/css/main.css:1]: dont add dist files in github!

File: dist/css/main.css [hunk 13]
```diff
@@ -0,0 +1,8 @@
+#header {
```
Comment (shreedharshetty) [inline dist/css/main.css]: why is ur css of dist folder, part of codebase? put this under gitignore

Comment (shreedharshetty) [inline dist/css/main.css]: also no id selectors please

File: server/routes.js [hunk 14]
```diff
@@ -0,0 +1,158 @@
...
+            } else {
+              req.session.cookie.expires = false;
+            }
+        res.redirect('/');
+    });
+
+};
+
+function merge_options(obj1,obj2){
+	'use strict';
```
Comment (shreedharshetty) [inline server/routes.js]: why use strict here?

File: src/css/main.css [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
```
Comment (shreedharshetty) [inline src/css/main.css:1]: why css? why not scss?

Comment (pranjalk) [reply src/css/main.css:1]: @shreedharshetty this is the build file.  @gautamvij this is not supposed to be here. add it in .gitignore

**Comment by witty123** [comment]
@gautamvij 
Add code-climate test.
Fix all the errors as pointed out by other reviewers.
Push your code and then peer review will start.

**Comment by witty123** [comment]
Add code-climate test.

File: .gitignore [hunk 16]
```diff
@@ -1,3 +1,3 @@
 .DS_Store
 node_modules/
-config.json
+config.json
```
Comment (witty123) [inline .gitignore]: Add a new line at the end of every file that has this "small red enter" icon at the end.

**Comment by pranjalk** [comment]
@gautamvij make a minor change and push to branch for code climate to run!

File: .gitignore [hunk 17]
```diff
@@ -1,3 +1,4 @@
 .DS_Store
 node_modules/
 config.json
+dist/
```
Comment (witty123) [inline .gitignore]: add new line

File: src/css/generic/_generic.scss [hunk 18]
```diff
@@ -0,0 +1,132 @@
...
+q:before,
+q:after {
+ content: '';
+ content: none;
+}
+
+table {
+ border-collapse: collapse;
+ border-spacing: 0;
+}
```
Comment (witty123) [inline src/css/generic/_generic.scss:132]: remove sass code for tags which are not being used. Follow ITCSS philosophy while writing sass.

File: src/css/main.css (hunk 15 above)
Comment (witty123) [reply src/css/main.css:1]: @gautamvij you still have main.css file in this commit. Remove from cache and filter from branch

**Pull request merged**